.circleci/config.yml

```diff
@@ -19,34 +19,28 @@ jobs:
 
       - run: sudo apt-get update && sudo apt-get -y install build-essential pkg-config autoconf gnutls-dev
       - run: ./install_ffmpeg.sh
-      - run: go get github.com/golang/glog
-      - run: go get -u -v github.com/ericxtang/m3u8
-      - run: go get github.com/aws/aws-sdk-go/aws
-      - run: go get -u google.golang.org/grpc
-      - run: go get github.com/jstemmer/go-junit-report
-      - run: go get github.com/pkg/errors
-      - run: go get github.com/stretchr/testify
-      - run: go get github.com/alecthomas/gometalinter && gometalinter --install
-      - run: go get -u -v go.opencensus.io/stats
-      - run: go get -u -v go.opencensus.io/tag
-      - run: go get -u -v contrib.go.opencensus.io/exporter/prometheus
+      - run: ./install_dependencies.sh
```

Comment:
If we're refactoring / cleaning up this part, I'd really much rather just `go get ./...` and fetch everything at once, since enumerating deps in this way feels brittle (eg, it's easy to lose track of what's actually required, so fresh installs could end up pulling down unnecessary packages).

I believe the results of `go get` are cached by golang after the first run, but it's tricky to determine when the cache needs to be updated. @darkdarkdragon mentioned that `go get -u ./...` is slow (since it then tries to fetch everything), but right now the packages that are `-u`'d are totally arbitrary (and it gradually slows things down the more we add).

One approach is to check in a file with a nonce or timestamp and cache that within CircleCI [1][2]. If we check in dependency changes, we can update the file. This script can pick that up and run `go get -u ./...` to update the deps. This step could be integrated as part of the Makefile build process as well.

[1] Just checked whether we could just `touch` and `stat` an empty file, but git doesn't seem to pick those up...

[2] In an ideal packaging world, this file would be a list of dependencies and a list of commit hashes corresponding to the version of the dependency we want. If anything on the list changes, the system would synchronize the local copy to the correct hashes. Probably too much trouble for us right now though?

Reply:
I seem to recall `go get -u ./...` not working at all; let me run a test and see if I'm remembering that properly.

Reply:
@j0sh @iameli I would rather not put any effort into optimizing this right now - I think it better to just to wait for the next `golang` release and switch to the modules.

> In an ideal packaging world, this file would be a list of dependencies and a list of commit hashes

Modules do just that